client/src/Components/Address/Address.elm

```diff
@@ -165,63 +181,265 @@ type Msg
+autocompletePreviewDecoder =
+    Decode.map3 AutocompletePreview
+        (Decode.field "status" Decode.string)
+        (Decode.field "message" Decode.string)
+        (Decode.field "data" (Decode.list autocompleteDataDecoder))
+
+
+fetchSuggestions : String -> String -> Cmd Msg
+fetchSuggestions str postgridApiKey =
+    let
+        request =
+            { method = "GET"
+            , headers = [Http.header "x-api-key" postgridApiKey]
+            , url = "https://api.postgrid.com/v1/addver/completions?partialStreet=" ++ (percentEncode str) ++ "&countryFilter=US&provInsteadOfPC=false"
+            , body = Http.emptyBody
+            , expect = Http.expectString Result.toMaybe
+            , timeout = Nothing
+            , tracker = Nothing
+            }
+    in
+        Api.sendRequest AutocompleteSuggestions (Api.withErrorHandler autocompletePreviewDecoder request)
+
+-- https://postgrid.readme.io/reference/autocomplete-an-address
+type alias AutocompleteResponse =
+    { status : String
+    , message : String
+    , data : List AutocompleteResponseData
+    }
+
+autocompleteResponseDecoder : Decoder AutocompleteResponse
+autocompleteResponseDecoder =
+    Decode.map3 AutocompleteResponse
+        (Decode.field "status" Decode.string)
+        (Decode.field "message" Decode.string)
+        (Decode.field "data" <| Decode.list autocompleteResponseDataDecoder)
+
+type alias AutocompleteResponseData =
+    { address : String
+    , city : String
+    , prov : String
+    , pc : String
+    , country : String
+    , errors : Maybe (List String)
+    }
+
+autocompleteResponseDataDecoder : Decoder AutocompleteResponseData
+autocompleteResponseDataDecoder =
+    let
+        errorsDecoder : Decoder (Maybe (List String))
+        errorsDecoder =
+            Decode.dict (Decode.list Decode.string)
+                |> Decode.andThen
+                    (\dictVal ->
+                        if Dict.isEmpty dictVal then
+                            Decode.succeed Nothing
+                        else
+                            case Dict.get "line1" dictVal of
+                                Just errs -> Decode.succeed (Just errs)
+                                Nothing -> Decode.fail "Expected 'line1' key in errors object"
+                    )
+    in
+    Decode.map6 AutocompleteResponseData
+        (Decode.at ["address", "address"] Decode.string)
+        (Decode.at ["address", "city"] Decode.string)
+        (Decode.at ["address", "prov"] Decode.string)
+        (Decode.at ["address", "pc"] Decode.string)
+        (Decode.at ["address", "country"] Decode.string)
+        (Decode.field "errors" errorsDecoder)
+
+completeAddress : AutocompleteData -> String -> Cmd Msg
+completeAddress suggestion postgridApiKey =
+    let
+        request =
+            { method = "POST"
+            , headers = [Http.header "x-api-key" postgridApiKey]
+            , url = "https://api.postgrid.com/v1/addver/completions"
```

Comment:
Why do we call this endpoint after a user select an address? I think we can use the selected value directly right 🤔

Reply:
The previous call only shows completion previews, while this endpoint actually completes the address. 
The same is basically done by postgrid frontend integration, see https://postgrid.readme.io/docs/address-autocomplete